lucille-core/src/main/java/com/kmwllc/lucille/util/SolrUtils.java

```diff
@@ -117,21 +119,58 @@ public static List<String> getSolrUrls(Config config) {
 
   /**
    * This method will convert a Tuple produced by a Solr Streaming Expression into a Lucille Document. The Tuple must
-   * have a value in its id field in order to be converted.
+   * have a value in its id field in order to be converted. The types of values are maintained in the conversion. Empty list 
+   * are not included in the created document
    *
    * @param tuple a Tuple from Solr
-   * @return a Document
+   * @throws DocumentException if {@param tuple} does not have a key called Document.ID_FIELD
+   * @return a Document whose id is the stringified version of the contents of Document.ID_FIELD in the tuple.
    */
   public static Document toDocument(Tuple tuple) throws DocumentException {
     Document d;
-    if (tuple.getString(Document.ID_FIELD) != null) {
+
+    if (tuple.getFields().containsKey(Document.ID_FIELD)) {
       d = Document.create(tuple.getString(Document.ID_FIELD));
     } else {
       throw new DocumentException("Unable to create Document from Tuple. No id field present.");
     }
 
     for (Entry<String, Object> e : tuple.getFields().entrySet()) {
-      d.setOrAdd((String) e.getKey(), (String) e.getValue());
+      Object val = e.getValue();
+      String key = e.getKey();
+
+      if (key.equals(Document.ID_FIELD)) {
+        continue;
+      }
+      DocumentException exception = new DocumentException(
```

Comment:
suggest we don't instantiate this exception unless we're actually going to throw it
I would just write this out on line 168 and 172 -- the little bit of duplication that would be introduced between those lines seems ok to me, but if you really wanted you could avoid it with a private static getException helper method.